clash_royale_service.py

```diff
@@ -168,4 +178,74 @@ def past_weeks_clan_war(self, clan_players_war_history: Dict[str, PlayerActivity
             player.war_active = compute_war_active(player.fame_hist)
             player.elder_worthy = compute_elder_worthy(player.fame_hist)
             player.avg_fame = compute_average_fame(player.fame_hist)
-        return clan_players_war_history
\ No newline at end of file
+        return clan_players_war_history
+
+    '''
+    Remaining war attacks
+    '''
+
+    def _get_participated(self, participants: List[Any]) -> int:
+        count = 0
+        for person in participants:
+            if person["decksUsedToday"] != 0:
+                count += 1
+        return count
+
+    def _get_decks_remaining(self, participants: List[Any], clan_members: List[Any]) -> int:
+        tag_to_player_war = {person["tag"]: person for person in participants}
+        tag_to_player_clan = [person["tag"] for person in clan_members]

```

Comment:
Ummmmm, shouldn't this be a set instead of a list? (Did I do this? my bad if so hahahaha)
For the `if player not in tag_to_player_clan` check,
- list is O(N) for lookup
- set is amortised O(1) for lookup

Reply:
you're right. i'll fix it for other lines too